projects/app/bin/generateRadicalNameMnemonics.ts

```diff
@@ -141,26 +121,27 @@ Use the following strategy:
 
 Write 10 mnemonic variations for ${hanzi} (${name}).
 `,
-      },
-    ],
-    response_format: {
-      type: `json_schema`,
-      json_schema: {
-        name: `radical_mnemonics`,
-        schema: {
-          type: `object`,
-          properties: {
-            radical: { type: `string` },
-            name: { type: `string` },
-            mnemonics: {
-              type: `array`,
-              items: {
-                type: `object`,
-                properties: {
-                  mnemonic: { type: `string` },
-                  reasoning_steps: {
-                    type: `array`,
-                    items: { type: `string` },
+        },
+      ],
+      response_format: {
```

Comment:
did you know that you can actually use a first-class Zod schema to drive this?

Like so:
```ts
const schema = z
  .object({
    classification: z.array(
      z
        .object({
          foo: z.enum(fooValues),
          bar: z.enum(barValues),
        })
        .strict(),
    ),
  })
  .strict();
type Schema = z.infer<typeof schema>;

const responseFormat = zodResponseFormat(
  schema,
  "my_result",
);
```
You can accept this in your own function via `OpenAI.Chat.Completions.ChatCompletionCreateParams.ResponseFormat`.

I can highly recommend it, makes keeping these calls much easier longterm.

Reply:
ah yeah this is good, i had done a quick google before but didn't find it and just put it on my todo to improve later